GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -222,19 +231,27 @@ private void ConfigureMessageBox()
 
         private void AssignCommitMessageFromTemplate()
         {
-            if (IsUICommandsInitialized)
+            if (!IsUICommandsInitialized)
             {
-                // Check if commit.template is used
-                string fileName = Module.GetEffectiveSetting("commit.template");
-                if (!string.IsNullOrEmpty(fileName))
-                {
-                    using (var commitReader = new StreamReader(fileName))
-                    {
-                        _commitTemplate = commitReader.ReadToEnd().Replace("\r", "");
-                    }
-                    Message.Text = _commitTemplate;
-                }
+                return;
+            }
+
+            try
+            {
+                Message.Text = _commitTemplate = _commitTemplateManager.LoadGitCommitTemplate();
+                return;
+            }
+            catch (FileNotFoundException ex)
+            {
+                MessageBox.Show(this, string.Format(_templateNotFound.Text, ex.FileName),
+                    _templateNotFoundCaption.Text, MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+            catch (Exception ex)
```

Comment:
should we handle `PathTooLongException` separately?

Reply:
It depends on how helpful you want to be. There can be also an exception related to invalid chars. 
If you want to be super helpful then I think `LoadGitCommitTemplate` should wrap its body in try-catch block and throw a new exception providing actual `commit.template` value and the caught exception's message. Then we could have a single try catch block here. But I am also fine with the existing solution.

Reply:
I leave it as is now, in necessary we can revisit it later.
The general exception handler will relay the error message to the user.